src/cryptoadvance/specter/util/descriptor.py

```diff
@@ -171,11 +187,19 @@ def parse(cls, desc, testnet=False):
                     keys.sort(key=lambda x: x if "]" not in x else x.split("]")[1])
             multisig_M = desc.split(",")[0].split("(")[-1]
             multisig_N = len(keys)
+            if int(multisig_M) > multisig_N:
```

Comment:
This addition ensures that no descriptor is parsed where less keys than the threshold are provided.  @stepansnigirev